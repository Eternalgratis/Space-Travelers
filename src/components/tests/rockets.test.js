import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Rockets from '../rockets/rockets';
import store from '../../redux/configureStore';
import rocketReducer, { fetchRockets } from '../../redux/rockets/rockets';

describe('Jest Snapshot testing suite', () => {
  it('matches DOM rockets snapshot', () => {
    const rocketDiv = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      );
    expect(rocketDiv).toMatchSnapshot();
  });

  it('testing for the newState to return empty array', () => {
    const newState = rocketReducer(undefined, {});
    expect(newState).toEqual([]);
  });
  it('the initialState should return empty array', () => {
    const initialState = rocketReducer([], {});
    expect(initialState).toEqual([]);
  });

  it('the rocketItem should equal values', () => {
    const rocketItems = fetchRockets();
    expect(rocketItems).not.toEqual([]);
  });
});
