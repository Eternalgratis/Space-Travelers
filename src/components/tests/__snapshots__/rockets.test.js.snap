// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite matches DOM rockets snapshot 1`] = `<div />`;
